pennylane/labs/resource_estimation/resource_tracking.py

```diff
@@ -0,0 +1,317 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Core resource tracking logic."""
+from collections import defaultdict
+from collections.abc import Callable
+from functools import singledispatch, wraps
+from typing import Dict, Iterable, List, Set, Union
+
+import pennylane as qml
+from pennylane.operation import Operation
+from pennylane.queuing import AnnotatedQueue
+from pennylane.tape import QuantumScript
+from pennylane.wires import Wires
+
+from .resource_container import CompressedResourceOp, Resources
+from .resource_operator import ResourceOperator
+
+# pylint: disable=dangerous-default-value,protected-access
+
+_StandardGateSet = {
```

Comment:
Are these dicts needed in different modules? If yes, safer to define them only once to prevent potential mismatch.

Reply:
They should only be in this module in this file.